internal/pkg/client/oci/ocisif.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2023 Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package oci
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/match"
+	ggcrmutate "github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/sylabs/oci-tools/pkg/mutate"
+	"github.com/sylabs/oci-tools/pkg/sif"
+	buildoci "github.com/sylabs/singularity/internal/pkg/build/oci"
+	"github.com/sylabs/singularity/internal/pkg/cache"
+	"github.com/sylabs/singularity/internal/pkg/util/fs"
+	"github.com/sylabs/singularity/pkg/sylog"
+	useragent "github.com/sylabs/singularity/pkg/util/user-agent"
+)
+
+// pull will create an OCI-SIF image in the cache if directTo="", or a specific file if directTo is set.
+//
+//nolint:dupl
```

Comment:
It's interesting that the linter here doesn't recognize that the two aren't actually perfect copies of one another (e.g. the string literals in the log & error msgs are different). That said, should we be pondering a refactor here...?